cmd/snap-confine/udev-support.c

```diff
@@ -197,6 +204,44 @@ void setup_devices_cgroup(const char *security_tag, struct snappy_udev *udev_s)
 	for (int i = 0; static_devices[i] != NULL; i++)
 		run_snappy_app_dev_add(udev_s, static_devices[i]);
 
+	// nvidia modules are proprietary and therefore aren't in sysfs and
+	// can't be udev tagged. For now, just add existing nvidia devices to
+	// the cgroup unconditionally (AppArmor will still mediate the access).
+	// We'll want to rethink this if snapd needs to mediate access to other
+	// proprietary devices.
+	struct stat sbuf;
+	char nv_path[15] = { 0 };	// /dev/nvidiaXXX
+	const unsigned nv_major = 195;
+	for (unsigned nv_minor = 0; nv_minor < 255; nv_minor++) {
+		// https://github.com/torvalds/linux/blob/master/Documentation/admin-guide/devices.txt
+		// /dev/nvidia0 through /dev/nvidia254
+		sc_must_snprintf(nv_path, sizeof(nv_path), "/dev/nvidia%u",
+				 nv_minor);
+
+		// Stop trying to find devices after one is not found. In this
+		// manner, we'll add /dev/nvidia0 and /dev/nvidia1 but stop
+		// trying to find nvidia3 - nvidia254 if nvidia2 is not found.
+		if (stat(nv_path, &sbuf) != 0) {
+			break;
+		}
+		_run_snappy_app_dev_add_majmin(udev_s, nv_path, nv_major,
+					       nv_minor);
+	}
+
+	const char *nvctl_path = "/dev/nvidiactl";
+	const unsigned nvctl_minor = 255;
```

Comment:
Could you please move that constant closer to `nv_major`? Alternatively let's chunk them all in `nvidia.h`.

Reply:
Done